processing/src/main/java/org/apache/druid/segment/join/JoinableFactory.java

```diff
@@ -19,13 +19,17 @@
 
 package org.apache.druid.segment.join;
 
+import com.google.common.annotations.Beta;
+import org.apache.druid.guice.annotations.ExtensionPoint;
 import org.apache.druid.query.DataSource;
 
 import java.util.Optional;
 
 /**
  * Utility for creating {@link Joinable} objects.
  */
+@Beta
```

Comment:
How does it make different from other extension points without `@Beta`? Does it mean this extension point can change in breaking ways even between minor releases?

Reply:
Yes. That's what I wanted to convey. It may also be that this ExtensionPoint goes away completely in a near future version, so core extensions should not rely on this.

Reply:
> It may also be that this ExtensionPoint goes away completely in a near future version, so core extensions should not rely on this.

Would you please add this to the javadoc?

> so core extensions should not rely on this.

Hmm, would you elaborate more on this? Are you referring to the Druid core extensions here? Per my understanding, the extension point and public api interfaces are more for third party extensions which are neither core nor contrib extensions. I think it doesn’t matter much whether the core or contrib extensions rely on some beta extension points or not because we will fix the code for them by ourselves. But the third party extensions are not in the Druid repository but in their own and so changed in extension points and apis should be noticed. 

How about making this interface as an extension point once all dusts settle down?

Reply:
Good suggestion. I thought it had to be an extension point for the factory to be bound from an extension. I've removed the annotations.